client/src/services/models/Product.js

```diff
@@ -0,0 +1,31 @@
+class Product {
+  constructor(id, farmerID, name, description, category) {
+    this.id = id;
+    this.farmerID = farmerID;
+    this.name = name;
+    this.description = description;
+    if (Object.values(Product.Categories).includes(category))
+      this.category = category;
+  }
+
+  static fromMongoJSON(json) {
```

Comment:
I don't know if this method should be called fromMongoJSON. It creates the object from the http response not from the MongoDB response.